lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -188,28 +188,24 @@ def cat_file(options, sha)
         @repo.lookup(sha).read_raw
       end
 
-      def revert(path, sha1, sha2)
-        options = {}
-        options[:location] = :index
-        files = []
-        if path
-          options[:paths] = [path]
-          patch = @repo.diff(sha2, sha1, options).first.diff
-          files << path
-        else
-          patch = @repo.diff(sha2, sha1)
-          patch.each_delta do |delta|
-            files << delta.new_file[:path]
-            files << delta.old_file[:path]
-          end
-        end
-        files.uniq!
+      def revert_path(path, sha1, sha2)
+        patch = @repo.diff(sha2, sha1).first.diff
         begin
-          @repo.apply(patch, options)
+          @repo.apply(patch, {:location => :index, :path => path)
         rescue RuntimeError
           return false
         end
-        return @repo.index.write_tree, files
+        return @repo.index.write_tree
+      end
+
+      def revert_commit(sha1, sha2)
```

Comment:
Should this method have sanity checks for its arguments (e.g., non-nil) or is that handled elsewhere?

Reply:
Added